presto-main/src/main/java/com/facebook/presto/server/GracefulShutdownHandler.java

```
@@ -61,23 +63,21 @@ public GracefulShutdownHandler(
             TaskManager sqlTaskManager,
             ServerConfig serverConfig,
             ShutdownAction shutdownAction,
-            LifeCycleManager lifeCycleManager)
+            LifeCycleManager lifeCycleManager,
+            QueryManager queryManager)
     {
         this.sqlTaskManager = requireNonNull(sqlTaskManager, "sqlTaskManager is null");
         this.shutdownAction = requireNonNull(shutdownAction, "shutdownAction is null");
         this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
         this.isCoordinator = requireNonNull(serverConfig, "serverConfig is null").isCoordinator();
         this.gracePeriod = serverConfig.getGracePeriod();
+        this.queryManager = queryManager;
```

use `requireNonNull` to fail fast.

---

Makes sense, I have made the changes